k9mail/src/main/java/com/fsck/k9/mailstore/LocalFolder.java

```diff
@@ -894,6 +894,42 @@ public LocalMessage doDbWork(final SQLiteDatabase db) throws WrappedException, U
         }
     }
 
+    public LocalMessage getMessageUidAndFlags(final String uid) throws MessagingException {
```

Comment:
No tests.

Reply:
I think it's a bit out of scope here. cketti mentioned that the tests for `LocalFolder` should test against a real SQLite db in https://github.com/k9mail/k-9/pull/2694#issuecomment-324682921